extension/mergeSort.js

```diff
@@ -0,0 +1,38 @@
+function mergeSort( initialSet = [] ){
+  if( initialSet == null ){
+    return 'Cannot sort null'
+  }
+  if( initialSet.constructor !== Array ){
+    return 'Please use Arrays'
+  }
+
+  const recurse = (arr) => {
+    let left = arr.slice(0, Math.floor(arr.length / 2))
+    let right = arr.slice(Math.floor(arr.length / 2), arr.length )
+
+    if(left.length > 1){
+      left = recurse(left)
+    }
+    if(right.length > 1){
+      right = recurse(right)
+    }
+
+    let resultArray = []
+    let iIndex = 0
+    let jIndex = 0
+
+    while(iIndex < left.length && jIndex < right.length){
```

Comment:
@deadlyicon Nested for loops was our initial approach, but we found it very difficult to actually get the desired effect.  And after some research, many other implementations we found worked similarly.

For the merge, you don't want to just iterate over one array for every item in the other array.  You're making a choice with each step which array to pick an item from, and then incrementing that array's index (or removing an item from that array).

(Also, come to think of it, doesn't a nested for loop suggest n^2 ?)